test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -52,161 +57,272 @@ class database_operation {
-            database.add_collection(collection_name);
-            testutil_check(
-              session->create(session, collection_name.c_str(), DEFAULT_FRAMEWORK_SCHEMA));
-            ts = timestamp_manager->get_next_ts();
-            tracking->save_schema_operation(
-              tracking_operation::CREATE_COLLECTION, collection_name, ts);
-        }
-        debug_print(std::to_string(collection_count) + " collections created", DEBUG_TRACE);
 
-        /* Open a cursor on each collection and use the configuration to insert key/value pairs. */
-        key_count = config->get_int(KEY_COUNT);
+        /* Get our configuration values, validating that they make sense. */
+        collection_count = config->get_int(COLLECTION_COUNT);
+        key_count = config->get_int(KEY_COUNT_PER_COLLECTION);
         value_size = config->get_int(VALUE_SIZE);
+        thread_count = config->get_int(THREAD_COUNT);
+        testutil_assert(collection_count % thread_count == 0);
         testutil_assert(value_size > 0);
         key_size = config->get_int(KEY_SIZE);
         testutil_assert(key_size > 0);
+
         /* Keys must be unique. */
         testutil_assert(key_count <= pow(10, key_size));
 
-        for (const auto &collection_name : database.get_collection_names()) {
-            key_cpt = 0;
-            /*
-             * WiredTiger lets you open a cursor on a collection using the same pointer. When a
-             * session is closed, WiredTiger APIs close the cursors too.
-             */
+        /* Create n collections as per the configuration and store each collection name. */
+        for (int64_t i = 0; i < collection_count; ++i) {
+            /* FIXME-T-F: Should we just give collection creation power to the database? */
+            collection_name = database.add_collection();
             testutil_check(
-              session->open_cursor(session, collection_name.c_str(), NULL, NULL, &cursor));
-            for (size_t i = 0; i < key_count; ++i) {
-                /* Generation of a unique key. */
-                generated_key = number_to_string(key_size, key_cpt);
-                ++key_cpt;
+              session->create(session, collection_name.c_str(), DEFAULT_FRAMEWORK_SCHEMA));
+            ts = timestamp_manager->get_next_ts();
+            tracking->save_schema_operation(
+              tracking_operation::CREATE_COLLECTION, collection_name, ts);
+            collection_names.push_back(collection_name);
+        }
+        debug_print(
+          "Populate: " + std::to_string(collection_count) + " collections created.", DEBUG_INFO);
+
+        /*
+         * This is a lambda that takes a set of collections to populate with keys. It will fail if
+         * the set of collections is not unique.
+         *
+         * This should be safe as any captured value is only read, not written to.
+         */
+        auto populate = [&](int thread_id, std::vector<std::string> pop_collections,
```

Comment:
Replace catch all capture with specific capture, or move into its own function?

Reply:
I'd say this is significant enough (a solid page worth of code) that it should be in a separate helper function.

Reply:
On a side note, prefer to use `const std::vector<std::string>&` to avoid memcpying stuff everywhere.